### PR TITLE
Remove psycopg2 dependency

### PR DESCRIPTION
**Pull request by Lordshinjo**
- It was only used to format client-side string named parameters, for a
relatively trivial operation
- Remove all mentions of it in documentation
- Remove 'namedparams' extras requirement

### CODE REVIEW

File: vertica_python/vertica/cursor.py [hunk 1]
```diff
@@ -459,11 +452,10 @@ def format_operation_with_parameters(self, operation, parameters, is_csv=False):
         return operation
 
     def format_quote(self, param, is_csv):
-        # TODO Make sure adapt() behaves properly
         if is_csv:
             return u'"{0}"'.format(re.escape(param))
         else:
-            return QuotedString(param.encode(UTF_8, self.unicode_error)).getquoted()
+            return u"'{0}'".format(param.replace(u"'", u"''"))
```
Comment (sitingren) [inline vertica_python/vertica/cursor.py]: There are regressions with this change:
```python
cursor.execute("CREATE TABLE tbl (col_1 VARCHAR)")
cursor.execute("INSERT INTO tbl VALUES (:val)", {'val': u'\\backslash data\\'})
cursor.execute("SELECT * FROM tbl")
print(cursor.fetchall())
# Previously with psycopg2: [['\\backslash data\\']]
# Now removed psycopg2: [['\x08ackslash data\\']]
```

File: vertica_python/tests/integration_tests/test_cursor.py [hunk 2]
```diff
@@ -382,6 +382,15 @@ def test_format_quote_unicode(self):
...
 
+    def test_execute_parameter(self):
+        with self._connect() as conn:
+            cur = conn.cursor()
+            val = u"".join(chr(i) for i in range(1, 128))
+            cur.execute("SELECT :val",
+                        parameters={"val": val},
+                        use_prepared_statements=False)
+            self.assertEqual([val], cur.fetchone())
+
```
Comment (sitingren) [inline vertica_python/tests/integration_tests/test_cursor.py]: I suggest this test to be defined under `SimpleQueryTestCase` class. `CursorTestCase` class contains tests that should be run for both use_prepared_statements=True/False. This change only affects use_prepared_statements=False behaviors and `SimpleQueryTestCase` class is the right place for tests that can run for only use_prepared_statements=False.

Thanks for reminding me there are confusions about test classes, I'll add comments to them.

**Comment by sitingren** [comment]
@Lordshinjo Thanks for your contribution! This is an improvement we'd really love to have.

However, there are regressions in your commits, so I cannot merge your PR right away. It would be better to add those corner cases into tests.

File: vertica_python/vertica/cursor.py (hunk 1 above)
Comment (Lordshinjo) [reply vertica_python/vertica/cursor.py]: Oh, in PostgreSQL's `PQescapeString`, backslashes are indeed also doubled, thanks for spotting that!

File: vertica_python/tests/integration_tests/test_cursor.py (hunk 2 above)
Comment (Lordshinjo) [reply vertica_python/tests/integration_tests/test_cursor.py]: Yes, I was confused by that, moved it to its proper place.

**Comment by sitingren** [comment]
@Lordshinjo Thank you for your quick response!
